drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -527,7 +529,25 @@ local function initialize_switch(driver, device)
     device:set_field(COMPONENT_TO_ENDPOINT_MAP_BUTTON, component_map, {persist = true})
   end
 
-  if num_switch_server_eps > 0 then
+  if #switch_eps > 0 and #button_eps > 0 then
+    if tbl_contains(STATIC_BUTTON_PROFILE_SUPPORTED, #button_eps) then
```

Comment:
I would set `profile_name = nil` here for clarity since this conditional block makes an assumption that `profile_name` is nil for this logic to work.

Reply:
I have refactored this section which makes this change unnecessary now I believe!